week2/homework/src/index.js

```diff
@@ -1,3 +1,129 @@
 'use strict';
 
 // Write the homework code in this file
+
+const fs = require('fs');
+
+const toDoFile = 'toDo.txt';
+
+function readFile() {
+    return new Promise(
+        (resolve, reject) => fs.readFile(toDoFile, (err, data) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(data.toString());
+            };
+        })
+    );
+}
+
+function appendFile(value) {
+    return new Promise(
+        (resolve, reject) => fs.appendFile(toDoFile, `${value}\n`, (err, data) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(data);
+            };
+        })
+    );
+}
+
+function writeFile(...val) {
+    return new Promise(
+        (resolve, reject) => fs.writeFile(toDoFile, `${val.join('\n')}`, (err, data) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(data);
+            };
+        })
+    );
+}
+
+function listToDo() {
+    readFile()  
+        .then(data => { data !== '' ? console.log(`\nTo - Dos: \n\n${data}`) : console.log('\nThere is nothing in the To-Do list!\n') })
+        .catch(console.error);
+}
+
+function resetToDo() {
+    writeFile('')
+        .then(() => console.log('\nWell Done! You deleted all the items in the list.\n'))
+        .catch(console.error);
+}
+
+function updateToDo() {
+    readFile()
+        .then((data) => {
+            let willUpdate = data.split('\n');
+            if (isNaN(input)) {
+                printHelp();
+            } else {
+                console.log(`\nDone!  "${willUpdate[input - 1]}" is updated as "${update}"\n`);
+                willUpdate[input - 1] = update;
+                writeFile(...willUpdate)   
+                    .then(() => readFile())
+                    .then(data => console.log(data));
+            };
+        })
+        .catch(console.error);
+}
+
+function removeToDo() {
+    readFile()
+        .then(data => {
+            let myData = data.split('\n');
+            myData.splice(input - 1, 1);
+            writeFile(...myData)
+                .then(() => readFile())
+                .then(data => console.log(data));
+        })
+        .catch(console.error);
+}
+
+function printHelp() {
+    console.log(`
+    HackYourFuture Node.js Week 2 - Lecture To-Do App
+
+    Options:
+
+    add    [to-do]                   add a to-do item
+    help                             show help text
+    remove [line index] + ['to-do']  remove to-do
```

Comment:
You probably meant `remove [line index]   remove to-do`.

Reply:
Yeah. I meant remove(line index).